lib/puppet/defaults.rb

```diff
@@ -1570,7 +1570,10 @@ def self.default_diffargs
     :pluginsync => {
       :default    => true,
       :type       => :boolean,
-      :desc       => "Whether plugins should be synced with the central server.",
+      :desc       => "Whether plugins should be synced with the central server. This setting is
```

Comment:
@joshcooper Ok, the part I still don't quite follow: will pluginsync be performed if using a cached catalog and `pluginsync=true` is set explicitly?